Sources/ResourceProvider/Cache/AnyAsyncCache.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  AnyAsyncCache.swift
+//  swift-resource-provider
+//
+//  Created by Óscar Morales Vivó on 4/11/23.
+//
+
+/**
+ Type-erased async cache.
+
+ This wrapper value type can be (and is) used to build up adapters for actual cache types, and can also be used to
+ build mocks for testing.
+ */
+public struct AnyAsyncCache<ID: Hashable, Value> {
```

Comment:
Also declare a `AnyAsyncCache: Sendable where ID: Sendable, Value: Sendable` extension. Basically all your public types will need a sensibility declaration.

Reply:
Yup that's NFTPR (but I want to do it Real Soon™).